README.md

```diff
@@ -50,7 +50,9 @@ Full list of options in `config.json`:
 
 | Property                                | Type      | Required?    | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
 | -------------------------------------   | --------- | ------------ | ---------------------------------------------------------------                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
-| project_id                              | String    | Yes          | BigQuery project                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
+| project_id                              | String    | Yes          | BigQuery project
+| gcs_bucket                              | String    | Yes          | Google Cloud Storage Bucket to use to stage files                                                             |
```

Comment:
can we make this optional instead? I'd prefer if we don't force people to use GCS as I know not everyone is using it and it's a breaking change.
Should be OK to upload then read from GCS if this parameter is provided or use `load_table_from_file` if not

Reply:
IMO there's no real reason not to ingest via GCS. It's certainly not a priority for us and I don't think I'd have the time to make the change + add additional testing in the near future. Though you're more than welcome to make the necessary changes. It will uglify the code a bit too.

Reply:
the thing is that we have been using it without uploading to GCS for years now and this would make it more difficult for us to have PII compliance unless we deleted the files later. I'm going to merge this to a new jsonl branch and I'll do that change and some testing on it

Reply:
This implementation does delete the files: https://github.com/jmriego/pipelinewise-target-bigquery/blob/c1272d384067477539ba3a62724b62e25f841074/target_bigquery/db_sync.py#L443-L446

Reply:
sounds good! just for the sake of conversation. Let's say we make GCS mandatory. What would be the benefits of doing that?
I can see that it would be more similar to the Snowflake target maintained by PipelineWise that forces you to use a stage or S3. But I don't think you'd win any speed, debug capabilities (if we delete the file in a finally) or anything else.
What do you think?

Reply:
![image](https://user-images.githubusercontent.com/17158624/155392107-ee57c3ed-0bf2-4487-86bc-327d1ad2c7c2.png)

from https://www.oreilly.com/library/view/google-bigquery-the/9781492044451/ch04.html

^ This suggests that when your network is fast loading into GCS is better. At least the way we're running this (and I'd guess is common) both our DB and (obviously) BQ are in GCP (in the same region in fact). So the network connection is very fast. This makes GCS the obvious choice.

Reply:
Also given that we're already not using compression that also suggests that outside GCP loading via GCS is still quicker.